src/NumericalAlgorithms/RootFinding/Secant.hpp

```diff
@@ -97,7 +97,7 @@ DataVector secant(const Function& f, const DataVector& lower_bound,
     // number of iterations that it took to find the root, so we reset it to
     // max_iterations after each root find.
     boost::uintmax_t max_iters = max_iterations;
-    auto result = boost::math::tools::toms748_solve(
+    auto result = boost::math::tools::toms748_solve(  // NOLINT
```

Comment:
add comment  for reason of NOLINT

Reply:
Actually I don't know the reason why clang-tidy fails for this function, it's an issue somewhere in boost. Indeed also the Travis check for clang-tidy still fails, despite the `NOLINT`, could you help me out with that?

Reply:
Clang-tidy fails because of something in Boost. Will looked into this at one point and clang-tidy considers this a "feature" (we don't since we can't fix code inside Boost), so the best we can do is offer an explanation and ignore the build failure. The result of the NOLINT is that rather than pointing into Boost header files it points to our invocation of Boost and so then people can read our comment.

Reply:
Ok I added comments. Now should I also `NOLINT` all calls to `RootFinder::toms748`?

Reply:
Just put back the comment you removed earlier in the PR.

Reply:
Ok done